src/Bicep.Core/Syntax/SeparatedSyntaxList.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Bicep.Core.Parser;
+
+namespace Bicep.Core.Syntax
+{
+    public class SeparatedSyntaxList : SyntaxBase
+    {
+        public SeparatedSyntaxList(IEnumerable<SyntaxBase> elements, IEnumerable<SyntaxBase> separators, TextSpan span)
```

Comment:
>SeparatedSyntaxList [](start = 15, length = 19)

This isn't used in this PR anymore, but leaving it for now. I think I will need it for function arguments.

Reply:
This class is an emotional rollercoaster